src/GameData.ts

```diff
@@ -0,0 +1,262 @@
+            case "item": {
+                result = this.findItem(args.slice(1).join(" "));
+                break;
+            }
+            case "perk": // fall through
+            case "rune": {
+                result = this.findPerk(args.slice(1).join(" "));
+                break;
+            }
+            case "champion": {
+                result = this.findChampion(args.slice(1).join(" "));
+                break;
+            }
+        }
+
+        if (typeof result === "string") {
+            message.channel.send(result);
+            return;
+        }
+
+        result.forEach(x => message.channel.send("```" + JSON.stringify(x, null, 4) + "```"));
+    }
+
+    public findItem(search: string): string | any[] {
+        if (!search) {
+            return `There are currently ${this.itemData.length} items in my lookup data!`;
+        }
+
+        // match name loosely, but ids strictly
+        const result = this.itemData.map(c => ({
+            value: c,
+            textscore: Math.min(
+                levenshteinDistance(search, c.name),
+                levenshteinDistanceArray(search, c.categories),
+            ),
+            idscore: Math.min(
+                levenshteinDistance(search, c.id.toString()),
+            ),
+        })).filter(x => x.textscore < this.sharedSettings.lookup.textConfidence || x.idscore < this.sharedSettings.lookup.numberConfidence);
+
+        if (result.length === 0) {
+            return "Unable to find any data for that search term.";
+        }
+
+        result.sort((a, b) => Math.min(a.textscore, a.idscore) > Math.min(b.textscore, b.idscore) ? 1 : -1);
+
+        if (result.length > this.sharedSettings.lookup.returnedEntryCount) {
+            let response = "Too many results returned for that search, did you mean one of the options below?\n";
+            response += result.map(x => `\`${x.value.name}\``).join("\n");
+            return response;
+        }
+
+        return result.slice(0, this.sharedSettings.lookup.returnedEntryCount).map(x => ({
+            id: x.value.id,
+            name: x.value.name,
+            combineCost: x.value.price,
+            cost: x.value.priceTotal,
+            from: x.value.from,
+            to: x.value.to,
+        }));
+    }
+
+    public findPerk(search: string): string | any[] {
+        if (!search) {
+            return `There are currently ${this.perkData.length} perks in my lookup data!`;
+        }
+
+        // match name loosely, but ids strictly
+        const result = this.perkData.map(c => ({
+            value: c,
+            textscore: Math.min(
+                levenshteinDistance(search, c.name),
+            ),
+            idscore: Math.min(
+                levenshteinDistance(search, c.id.toString()),
+            ),
+        })).filter(x => x.textscore < this.sharedSettings.lookup.textConfidence || x.idscore < this.sharedSettings.lookup.numberConfidence);
+
+        if (result.length === 0) {
```

Comment:
This function seems to work unexpected: (Ignore the savagery request)
![image](https://user-images.githubusercontent.com/3397066/55379746-49613d00-551e-11e9-9ea6-0ad7f18fa9be.png)

If the request is not exact, it gives no results.

Expected is something alike this:
![image](https://user-images.githubusercontent.com/3397066/55379785-6564de80-551e-11e9-9f7f-03f3050be898.png)

Reply:
The text match is set to match up-to 2 different characters, so it wont match the other ones unless its changed to a larger number. Might want to separate it for champs / runes / items, since champs have relatively shorter names.